aeppl/abstract.py

```diff
@@ -15,6 +16,26 @@ class MeasurableVariable(abc.ABC):
 MeasurableVariable.register(RandomVariable)
 
 
+class UnmeasurableMeta(MetaType):
+    def __eq__(self, other):
+        if hasattr(self, "id_obj") and hasattr(other, "id_obj"):
+            return hash(self.id_obj) == hash(other.id_obj)
```

Comment:
Can you do an `issubclass(other, UnmeasurableMeta)` check instead, or something like it?  That, as well as adding/requiring `id_obj` in a `__new__` implementation would make these `__eq__` and `__hash__` implementations more direct, especially since we don't want `x == y` to be true for any `x` and `y` that have equal `id_obj` attributes.

Reply:
Absolutely. The correct check was `isinstance(other, UnmeasurableMeta)`